code/modules/projectiles/pins.dm

```diff
@@ -222,4 +226,99 @@ Pins Below.
+	wireless_firing_pins -= src
+	return ..()
+
+/*
+	The return of this pin_auth is dependent on the wireless pin's lockstatus. For the purposes of this, only energy projectiles
+	that have a taser_effect are considered "stun" projectiles. All ballistic firearms are considered as being lethal, regardless
+	of ammunition loaded. Behaviours should be as follows:
+	Automatic: The weapon will only fire on stun while the security level is Green or Blue. On Yellow, Red & Delta alert the weapon is unrestricted
+	Disabled: The weapon will not fire under any circumstance.
+	Stun-Only: The weapon will only fire on stun regardless of the current security level.
+	Lethal: The weapon will fire on all modes regardless of the current security level.
+*/
+/obj/item/device/firing_pin/wireless/pin_auth(mob/living/user)
+	if(lockstatus != WIRELESS_PIN_DISABLED)
+		if(lockstatus == WIRELESS_PIN_LETHAL)
+			return TRUE
+		if(istype(gun, /obj/item/gun/energy))
+			var/obj/item/gun/energy/thegun = gun
+			var/obj/item/projectile/energy/P = new thegun.projectile_type
+			if(P?.taser_effect)
+				return TRUE
+			if(security_level != SEC_LEVEL_GREEN && security_level != SEC_LEVEL_BLUE && lockstatus == WIRELESS_PIN_STUN)
+				if(!P?.taser_effect)
+					return FALSE
+
+		if(security_level != SEC_LEVEL_GREEN && security_level != SEC_LEVEL_BLUE && lockstatus != WIRELESS_PIN_STUN)
+			return TRUE
+	return FALSE
+
+/obj/item/device/firing_pin/wireless/proc/unlock(var/i) // Changes the current lockstatus of the weapon, and sends a message and sfx to whoever is holding it.
+	var/mob/user = get_holding_mob(src)
+
+	if(i == lockstatus)
+		return
+
+	if(i == WIRELESS_PIN_AUTOMATIC)
+		playsound(user, 'sound/weapons/laser_safetyon.ogg')
+		to_chat(user, SPAN_NOTICE("<b>\The [gun]'s wireless-control firing pin is now set to automatic.</b>"))
+		lockstatus = WIRELESS_PIN_AUTOMATIC
+
+	if(i == WIRELESS_PIN_DISABLED)
+		playsound(user, 'sound/weapons/laser_safetyoff.ogg')
+		to_chat(user, SPAN_WARNING("<b>\The wireless-control firing pin locks \the [gun]'s trigger!</b>"))
+		lockstatus = WIRELESS_PIN_DISABLED
+
+	if(i == WIRELESS_PIN_STUN)
+		playsound(user, 'sound/weapons/laser_safetyon.ogg')
+		to_chat(user, SPAN_NOTICE("<b>\The [gun]'s wireless-control firing pin is now set to stun only.</b>"))
+		lockstatus = WIRELESS_PIN_STUN
+
+	if(i == WIRELESS_PIN_LETHAL)
+		playsound(user, 'sound/weapons/laser_safetyon.ogg')
+		to_chat(user, SPAN_NOTICE("<b>\The [gun]'s wireless-control firing pin is now unrestricted.</b>"))
+		lockstatus = WIRELESS_PIN_LETHAL
+	return
+
+/obj/item/device/firing_pin/wireless/attackby(obj/item/C as obj, mob/user as mob) //Lets people register their IDs to the pin. Using it once registers you, using it again clears you.
+	if(istype(C, /obj/item/card/id))
+		var/obj/item/card/id/idcard = C
+		if(idcard.registered_name == registered_user)
+			to_chat(user, SPAN_NOTICE("You press your ID against the RFID reader and it deregisters your identity."))
```

Comment:
Anyone can de-register a gun. When they use their ID on it, it will overwrite any existing identity with their own, and then using their ID again will clear it.